### DISCUSSION TITLE
How to sort buffers?

### DISCUSSION THREAD

**Discussion by netjune** [minad/vertico#397] [Q&A] [answered]
Let buffers whose name has prefix ' ' or '*' stay at the last of the completion menu.

**Comment by netjune** [comment]
It will be better if behaves like the ido-mode. In ido-mode, it hides these buffers initially when completing. If you want to see them in completing, you can press a keybind(like `C-a`). Then the completion menu is updated, and the hidden buffer are shown now.

**Reply by minad** [reply, marked as answer]
If you configure the basic completion style, then `switch-to-buffer` behaves like this. The hidden buffers are hidden at first and only displayed if you enter a SPC as first character.

~~~ elisp
;; For testing, ensure that only `basic` is used
(setq completion-styles '(basic)
      completion-category-defaults nil
      completion-category-overrides nil)
~~~

`consult-buffer` behaves like this for *all* completion styles, and also allows you to customize the buffers which are hidden via `consult-buffer-filter`.

In case you want to override sorting there are also the variables `vertico-sort-function` and `vertico-sort-override-function`, which can be configured via `vertico-multiform-mode` as documented in the README.
